src/algorithms/synthesis/encoding.cpp

```diff
@@ -93,13 +103,41 @@ namespace syrec {
             output.resize(requiredGarbage);
         }
 
+        // modify the codewords by filling in the redundant dc positions.
+        for (auto& [pattern, code]: encoding) {
+            TruthTable::Cube outCube(pattern);
+            outCube.resize(nBits);
+
+            TruthTable::Cube newCode{};
+            newCode.reserve(requiredGarbage);
+            for (auto i = 0U; i < requiredGarbage; i++) {
+                if (code[i].has_value()) {
+                    newCode.emplace_back(code[i]);
+                } else {
+                    newCode.emplace_back(outCube[r + i]);
+                }
+            }
+
+            encoding[pattern] = newCode;
+        }
+
         // encode all the outputs
         for (auto& [input, output]: tt) {
             output = encoding[output];
         }
+
+        return std::make_tuple(encoding, additionalLines);
     }
 
-    auto augmentWithConstants(TruthTable& tt) -> void {
+    auto augmentWithConstants(TruthTable& tt, std::size_t const& nBits) -> void {
+        //add necessary constant inputs to the outputs based on the total number of bits (nBits)
+        const auto ancillaBits = nBits - tt.nOutputs();
+        for (auto& [input, output]: tt) {
+            for (auto i = 0U; i < ancillaBits; i++) {
+                output.insertZero();
+            }
+        }
```

Comment:
couldn't this be integrated into the loop down below in order to avoid iterating over the complete truth table twice?